File: core/src/nl/t64/game/rpg/components/conversation/ConversationPhrase.java
package nl.t64.game.rpg.components.conversation;

import lombok.Getter;

import java.util.List;


@Getter
public class ConversationPhrase {

    private String face;
    private List<String> text;
    private List<ConversationChoice> choices;

    private ConversationPhrase() {
        this.face = "";
        this.text = null;   // mandatory in Json file
        this.choices = List.of();
    }

    List<ConversationChoice> getChoices(String currentPhraseId) {
        if (choices.isEmpty()) {
            return createArrowChoiceThatPointsToNextPhraseId(currentPhraseId);
        } else {
            return getVisibleChoices();
        }
    }

    private List<ConversationChoice> createArrowChoiceThatPointsToNextPhraseId(String currentPhraseId) {
        String nextDestinationId = String.valueOf(Integer.parseInt(currentPhraseId) + 1);
        return List.of(new ConversationChoice(nextDestinationId));
    }

    private List<ConversationChoice> getVisibleChoices() {
        return choices.stream()
                      .filter(ConversationChoice::isVisible)
                      .toList();
    }

}

File: core/src/nl/t64/game/rpg/components/conversation/ConversationPhrase.kt
package nl.t64.game.rpg.components.conversation


class ConversationPhrase(
    val face: String = "",
    val text: List<String> = emptyList(),   // mandatory in Json file
    val choices: List<ConversationChoice> = emptyList()
) {
    fun getChoices(currentPhraseId: String): List<ConversationChoice> {
        return if (choices.isEmpty()) {
            createArrowChoiceThatPointsToNextPhraseId(currentPhraseId)
        } else {
            getVisibleChoices()
        }
    }

    private fun createArrowChoiceThatPointsToNextPhraseId(currentPhraseId: String): List<ConversationChoice> {
        val nextDestinationId = (currentPhraseId.toInt() + 1).toString()
        return listOf(ConversationChoice(destinationId = nextDestinationId))
    }

    private fun getVisibleChoices(): List<ConversationChoice> = choices.filter { it.isVisible() }

}
